### PR TITLE
added support for spool manager plugin

### PR DESCRIPTION
**Pull request by WyattNielsen**
changes by  @abrenoch, based on an open PR in the abandoned repository. 

These changes have been running on my system for the past 2 weeks with no issues.

### CODE REVIEW

File: octoprint_costestimation/static/js/costestimation.js [hunk 1]
```diff
@@ -14,15 +14,18 @@ $(function() {
...
             return self.settings.settings.plugins.costestimation.requiresLogin() ?
                 self.loginState.isUser() : true;
         });
 
         self.showFilamentGroup = ko.pureComputed(function() {
-            return self.filamentManager === null || !self.settings.settings.plugins.costestimation.useFilamentManager();
-        })
+            var filamentManagerEnabled = self.filamentManager === null || !self.settings.settings.plugins.costestimation.useFilamentManager();
+            var spoolManagerEnabled = self.spoolManager === null || !self.settings.settings.plugins.costestimation.useSpoolManager();
+            return !filamentManagerEnabled && !spoolManagerEnabled;
```
Comment (somehibs) [inline octoprint_costestimation/static/js/costestimation.js:27]: `filamentManagerEnabled` should be `filamentManagerDisabled`.

**Comment by newphreak** [comment]
Can we get this going? :) I tested it for some time earlier, and it works fine.

**Comment by JeremyKennedy** [comment]
Please don't mark this stale, this is a feature I've been waiting on for a while now!

**Comment by newphreak** [comment]
> Please don't mark this stale, this is a feature I've been waiting on for a while now!

Me too

File: octoprint_costestimation/static/js/costestimation.js (hunk 1 above)
Comment (WyattNielsen) [reply octoprint_costestimation/static/js/costestimation.js]: Fix has been merged into PR, can you review again?

**Comment by Cjkeenan** [comment]
> Please don't mark this stale, this is a feature I've been waiting on for a while now!
I too am looking forward to this feature. How can I "sideload" it before the official pr goes through?

**Comment by abrenoch** [comment]
> > Please don't mark this stale, this is a feature I've been waiting on for a while now!
> 
> I too am looking forward to this feature. How can I "sideload" it before the official pr goes through?

You can use this zip, just install 'from URL' in the plugins dialog.
https://github.com/abrenoch/OctoPrint-CostEstimation/archive/spool-manager.zip

Octoprint may nag you to update though.

**Comment by JeremyKennedy** [comment]
I tried that link but it's a slightly older version and doesn't properly support Python 3. I've merged this PR on top of this repo so that we can get the latest changes from each branch - spool manager support, and Python 3 support. 

You can use this link to install it: https://github.com/JeremyKennedy/OctoPrint-CostEstimation/archive/master.zip

**Comment by OllisGit** [comment]
Hi all,

thanks for your effort, I appreciate that!!!!

In the newest version 3.2.0 the PR is included.

PS. Sorry, for being late to the party, but I made a "Octoprint sabatical" (no coding, no printing) the last months.

**Pull request closed without merging**